### ISSUE TITLE
Scenario not retaining orchestrate action resource order if you re-run using result.yml and it starts from a failed orchestrate task

### ISSUE THREAD

**Issue by Dannyb48** [RedHatQE/teflo#93]
I have a scenario where the first action task passed but the second action task failed. 

teflo_log:
```
21-06-29 20:23:09,864 INFO Starting tasks on pipeline: notify
2021-06-29 20:23:09,864 WARNING ... no tasks to be executed ...
2021-06-29 20:23:09,865 INFO ..................................................
2021-06-29 20:23:09,865 INFO Starting tasks on pipeline: orchestrate
2021-06-29 20:23:09,866 INFO --> Blaster v0.4.0 <--
2021-06-29 20:23:09,866 INFO Task Execution: Sequential
2021-06-29 20:23:09,867 INFO Tasks:
2021-06-29 20:23:09,867 INFO 1. Task     : Install OpenShift on libvirt
                                Class    : <class 'teflo.tasks.orchestrate.OrchestrateTask'>
                                Methods  : ['run']
2021-06-29 20:23:09,868 INFO 2. Task     : Install Openshift Container Storage on libvirt
                                Class    : <class 'teflo.tasks.orchestrate.OrchestrateTask'>
                                Methods  : ['run']

.
.
2021-06-29 21:56:49,713 INFO Successfully completed playbook : ansible/install_ocp_libvirt.yml
2021-06-29 21:56:49,753 INFO Orchestration passed : Successfully completed orchestrate action: Install OpenShift on libvirt.
2021-06-29 21:56:49,755 INFO    running orchestration Install Openshift Container Storage on libvirt for ['baremetal']
2021-06-29 21:57:02,421 ERROR Orchestration failed : Playbook ansible/install_ocs_libvirt.yml failed to run
```

teflo_results.yml:
```
orchestrate:
- name: Install OpenShift on libvirt
  description: null
  orchestrator: ansible
  hosts:
  - baremetal
  ansible_playbook:
    name: ansible/install_ocp_libvirt.yml
  ansible_options:
    extra_vars:
      artifact_directory: /home/dbaez/.virtualenvs/gs-workspace/cnv_odf_ocp/.teflo/.results/artifacts/sharedArtifacts
  ansible_galaxy_options:
    role_file: requirements.yml
  labels: []
  status: 0
- name: Install Openshift Container Storage on libvirt
  description: null
  orchestrator: ansible
  hosts:
  - baremetal
  ansible_playbook:
    name: ansible/install_ocs_libvirt.yml
  ansible_galaxy_options:
    role_file: requirements.yml
  labels: []
  status: 1


```

Then if I try to re-run from the failed task `teflo -s .teflo/.results/results.yml -t orchestrate` it starts what it is supposed to 
```
2021-06-29 22:02:18,601 INFO Starting tasks on pipeline: orchestrate
2021-06-29 22:02:18,602 INFO --> Blaster v0.4.0 <--
2021-06-29 22:02:18,602 INFO Task Execution: Sequential
2021-06-29 22:02:18,603 INFO Tasks:
2021-06-29 22:02:18,603 INFO 1. Task     : Install Openshift Container Storage on libvirt
                                Class    : <class 'teflo.tasks.orchestrate.OrchestrateTask'>
                                Methods  : ['run']
2021-06-29 22:02:18,604 INFO ** BLASTER BEGIN **
2021-06-29 22:02:18,606 INFO    running orchestration Install Openshift Container Storage on libvirt for ['baremetal']
.
.
2021-06-29 22:02:31,143 ERROR Orchestration failed : Playbook ansible/install_ocs_libvirt.yml failed to run
```

Granted it failed again but when I look at the results.yml I see this task is first rather than second
```
orchestrate:
- name: Install Openshift Container Storage on libvirt
  description: null
  orchestrator: ansible
  hosts:
  - baremetal
  ansible_playbook:
    name: ansible/install_ocs_libvirt.yml
  ansible_galaxy_options:
    role_file: requirements.yml
  labels: []
  status: 1
- name: Install OpenShift on libvirt
  description: null
  orchestrator: ansible
  hosts:
  - baremetal
  ansible_playbook:
    name: ansible/install_ocp_libvirt.yml
  ansible_options:
    extra_vars:
      artifact_directory: /home/dbaez/.virtualenvs/gs-workspace/cnv_odf_ocp/.teflo/.results/artifacts/sharedArtifacts
  ansible_galaxy_options:
    role_file: requirements.yml
  labels: []
  status: 0

```

This becomes a problem because after I fixed whatever was breaking my playbook in the second orchestrate task, the original first task ran again and overwrote what I had done with the second task.

I suspect there is some type of regression in the reload_resources code in the Scenario resource.

**Comment by rujutashinde** [comment]
Jira ticket CCITCARBON-333 has been created for this issue. Url : https://projects.engineering.redhat.com/browse/CCITCARBON-333

**Comment by rujutashinde** [comment]
fixed with #110
